build.gradle.kts

```diff
@@ -31,61 +34,140 @@ plugins {
     id("com.goncalossilva.resources") version "0.2.2" // kotlinx-resources
 }
 
-repositories { mavenCentral() }
+repositories {
+    mavenCentral()
+    google()
+}
 
 @Suppress("UNUSED_VARIABLE")
 kotlin {
     jvm {
         compilations.all { kotlinOptions.jvmTarget = "1.8" }
-        withJava()
         tasks.withType<Test> { useJUnitPlatform() }
     }
 
-    js(IR) { nodejs {} }
-
-    val hostOs = System.getProperty("os.name")
-    val isMingwX64 = hostOs.startsWith("Windows")
-    val nativeTarget =
-        when {
-            hostOs == "Mac OS X" -> macosX64("native")
-            hostOs == "Linux" -> linuxX64("native")
-            isMingwX64 -> mingwX64("native")
-            else -> throw GradleException("Host OS is not supported in Kotlin/Native.")
-        }
+    // Note: Android is configured separately below.
+    ios()
```

Comment:
Also, it looks like we are using 1.7.10 for Kotlin, which prevents consuming this library below Kotlin 1.7.10 on Kotlin Native. It will give this warning:

```
 Task :library-sync:compileKotlinIosArm64 FAILED
w: skipping /Users/cm/.gradle/caches/modules-2/files-2.1/org.mongodb/kbson-iosarm64/1.0-SNAPSHOT/9809a26692a34e1f22ad1f3b9459494d43262f83/kbson-iosarm64-1.0-SNAPSHOT.klib. Incompatible abi version. The current default is '1.6.0', found '1.7.0'. The library produced by 1.7.10 compiler
e: Could not find "/Users/cm/.gradle/caches/modules-2/files-2.1/org.mongodb/kbson-iosarm64/1.0-SNAPSHOT/9809a26692a34e1f22ad1f3b9459494d43262f83/kbson-iosarm64-1.0-SNAPSHOT.klib" in [/Users/cm/Realm/realm-kotlin/packages/library-sync, /Users/cm/.konan/klib, /Users/cm/.konan/kotlin-native-prebuilt-macos-aarch64-1.6.10/klib/common, /Users/cm/.konan/kotlin-native-prebuilt-macos-aarch64-1.6.10/klib/platform/ios_arm64]
```

This is a problem as Realm Kotlin currently ships with minimum version support of 1.6.10.

JetBrains currently do not promise ABI stability between Kotlin Native releases, but you can consume libraries from earlier versions. This is why libraries should just try to go as low as possible.

The only case where it might cause issues is because of our Serialization dependency, which needs to use an older version (that is compatible with Kotlin 1.6.1), which might mean that certain bugs haven't been fixed. But it is worth a try IMO.

Reply:
Interesting does the `iosSimulatorArm64` version provide something different over the `iosArm64` version?

Reply:
Will look into the kotlin version - not sure if that will be just an easy fix or uncover some KMM issues.

Reply:
The reason we need the `iosSimulatorArm64` target is due to how Kotlin creates the XCode frameworks (as far as I understand). The details are here: https://youtrack.jetbrains.com/issue/KT-43667

It then also matters because Gradle will attempt to match the target people are running on to a specific dependency. Kotlin produces maven metadata to do this matching, which is why you are also seeing a of lof publications, and if we don't have publications for all targets people might not be able to compile their projects.

Reply:
Thanks @cmelchior will see if I can get evergreen to publish that variant as well.